### PR TITLE
(PC-28860) feat(offer): fix movie calendar UI

### PR DESCRIPTION
**Pull request by xlecunff-pass**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-28860

## Checklist

I have:

- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [x] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]

## Screenshots

| Platform         | Mockup/Before                                                                                                                                                                                                                              | After                                                                                                                                                                                                              |
| :--------------- | :-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| Desktop - Chrome | <img width="914" alt="Capture d’écran 2024-03-29 à 16 53 49" src="https://github.com/pass-culture/pass-culture-app-native/assets/163895831/863d0b39-288d-40a7-91fa-266718c15b52"> | <video width="914" alt="After" src="https://github.com/pass-culture/pass-culture-app-native/assets/163895831/e54d85b5-2254-48b6-8980-f5dfbe69d491"> |
    
## Added behaviors

- ~~The components is scrollable on web~~
- The component takes the full right size on web
- A white fade out is added to both extremities of the component on web

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

**Review by thconte — APPROVED** [review]
LGTM

File: src/features/offer/components/MovieCalendar/MovieCalendar.tsx [hunk 1]
```diff
@@ -69,4 +74,30 @@ export const MovieCalendar: React.FC<Props> = ({
   )
 }
 
-const flatListContainer = { paddingHorizontal: getSpacing(6) }
+const flatListContainer = {
+  paddingHorizontal: getSpacing(6),
+}
+
+const FadeLeft = styled(LinearGradient).attrs({
```
Comment (NolwenMajorFrances) [inline src/features/offer/components/MovieCalendar/MovieCalendar.tsx:81]: J'ai essayé de voir si y avait moyen de moins répéter mais c'est pas foufou ahah :

 const createFadeComponent = (direction) => {
  let colors, start, end;
  
  switch (direction) {
    case 'left':
      colors = ['white', 'transparent'];
      start = { x: 0.2, y: 0.5 };
      end = { x: 1, y: 0.5 };
      break;
    case 'right':
      colors = ['transparent', 'white'];
      start = { x: 0, y: 0.5 };
      end = { x: 0.8, y: 0.5 };
      break;
   );
  }

  return styled(LinearGradient).attrs({
    colors: colors,
    start: start,
    end: end,
  })`
    position: absolute;
    top: 0;
    bottom: ${getSpacing(1)}px;
    ${direction}: -1px;
    width: ${getSpacing(20)}px;
  `;
};

const FadeLeft = createFadeComponent('left');
const FadeRight = createFadeComponent('right');

Comment (NolwenMajorFrances) [reply src/features/offer/components/MovieCalendar/MovieCalendar.tsx:81]: Sinon :
 const FadeComponent = styled(LinearGradient)`
  position: absolute;
  top: 0;
  bottom: ${getSpacing(1)}px;
  width: ${getSpacing(20)}px;
`;

const FadeLeft = FadeComponent.attrs({
  colors: ['white', 'transparent'],
  start: { x: 0.2, y: 0.5 },
  end: { x: 1, y: 0.5 },
})`
  left: -1px;
`;

const FadeRight = FadeComponent.attrs({
  colors: ['transparent', 'white'],
  start: { x: 0, y: 0.5 },
  end: { x: 0.8, y: 0.5 },
})`
  right: -1px;
`;

File: src/features/offer/components/MovieCalendar/MovieCalendar.tsx [hunk 2]
```diff
@@ -44,19 +45,23 @@ export const MovieCalendar: React.FC<Props> = ({
...
+          contentContainerStyle={flatListContainer}
+          showsHorizontalScrollIndicator={false}
+          onScroll={onScroll}
+          onContentSizeChange={onContentSizeChange}
+          testID="movie-calendar-flat-list"
+          renderItem={({ item: date }) => (
+            <MovieCalendarDay date={date} selectedDate={selectedDate} onTabChange={onTabChange} />
+          )}
+        />
+        {isDesktopViewport ? <FadeLeft /> : null}
```
Comment (tanguyprache) [inline src/features/offer/components/MovieCalendar/MovieCalendar.tsx:62]: Je pense qu'on peut regrouper les deux ici - tu en penses quoi ? : 
`{!!isDesktopViewport && 
(
          <React.Fragment>
            <FadeLeft />
            <FadeRight />
          </React.Fragment>
        )}`

**Review by tanguyprache — APPROVED** [review]

**Pull request merged**